packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -311,11 +313,27 @@ export const useArbTokenBridge = (
     } else {
       // looks like l1 address was provided
       l1Address = lowercasedErc20L1orL2Address
-      l2Address = await getL2ERC20Address({
-        erc20L1Address: l1Address,
-        l1Provider: l1.provider,
-        l2Provider: l2.provider
-      })
+
+      // while deriving the child-chain address, it can be a teleport transfer too, in that case derive L3 address from L1 address
+      // else, derive the L2 address from L1 address OR L3 address from L2 address
+      if (
+        isTeleport({
+          sourceChainId: l1.network.id,
+          destinationChainId: l2.network.id
+        })
+      ) {
+        l2Address = await getL3ERC20Address({
```

Comment:
This is quite hard to read because of `l1` and `l2` - they refer to parent/child chains but then we have things like `l1Provider` and `l3Provider`.

But could be fixed as a part of renaming refactoring I guess.

Reply:
Yeah true, I have expanded the comments present in this part of the code. But it's ultimate resolution will come when we rename `l2Address` to `childChainAddress`